Sources/App/Models/Manifest.swift

```diff
@@ -56,18 +74,21 @@ struct Manifest: Decodable, Equatable {
 }
 
 
-extension Manifest.Product.`Type`: Decodable {
+extension Manifest.ProductType: Decodable {
     init(from decoder: Decoder) throws {
-        let container = try decoder.container(keyedBy: Self.self)
-        for k in Self.allCases {
-            if let _ = try? container.decodeNil(forKey: k) {
-                self = k
-                return
-            }
-            
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        guard let key = container.allKeys.first(where: container.contains) else {
+            throw DecodingError.dataCorrupted(.init(codingPath: decoder.codingPath, debugDescription: "Did not find a matching key"))
+        }
+        switch key {
+            case .executable:
+                self = .executable
+            case .library:
+                var unkeyedValues = try container.nestedUnkeyedContainer(forKey: key)
+                let value = try unkeyedValues.decode(Manifest.LibraryType.self)
+                self = .library(value)
+            case .test:
+                self = .test
```

Comment:
`dump-package` uses a different JSON format for its associated values (one that's not Swift 5.5 forward compatible) and we need to have our own custom decoding for it.